### PR TITLE
Use correct defaults when binding apiserver flags

### PR DESCRIPTION
**Pull request by liggitt**
defaults should be set in the struct-creating function, then the current struct field value used as the default when binding the flag

### CODE REVIEW

File: cmd/kube-apiserver/app/options/options.go [hunk 1]
```diff
@@ -242,17 +245,17 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
...
-	fs.IntVar(&s.MaxRequestsInFlight, "max-requests-inflight", 400, "The maximum number of requests in flight at a given time.  When the server exceeds this, it rejects requests.  Zero for no limit.")
-	fs.IntVar(&s.MinRequestTimeout, "min-request-timeout", 1800, "An optional field indicating the minimum number of seconds a handler must keep a request open before timing it out. Currently only honored by the watch request handler, which picks a randomized value above this number as the connection timeout, to spread out load.")
+	fs.BoolVar(&s.EnableWatchCache, "watch-cache", s.EnableWatchCache, "Enable watch caching in the apiserver")
+	fs.BoolVar(&s.EnableSwaggerUI, "enable-swagger-ui", s.EnableSwaggerUI, "Enables swagger ui on the apiserver at /swagger-ui")
+	fs.StringVar(&s.ExternalHost, "external-hostname", s.ExternalHost, "The hostname to use when generating externalized URLs for this master (e.g. Swagger API Docs.)")
+	fs.IntVar(&s.MaxRequestsInFlight, "max-requests-inflight", s.MaxRequestsInFlight, "The maximum number of requests in flight at a given time.  When the server exceeds this, it rejects requests.  Zero for no limit.")
```
Comment (lavalamp) [inline cmd/kube-apiserver/app/options/options.go]: Is there not a separate place where the generic flags are added? If not I'll file an issue.

**Comment by lavalamp** [comment]
LGTM

File: cmd/kube-apiserver/app/options/options.go (hunk 1 above)
Comment (liggitt) [inline cmd/kube-apiserver/app/options/options.go]: not that I know of, I just tracked this back to the source and made sure no other sites were affected

Comment (lavalamp) [inline cmd/kube-apiserver/app/options/options.go]: k. I filed #24062

**Comment by liggitt** [comment]
rebased

**Pull request merged**